press/press/doctype/app_release/app_release.py

```diff
@@ -101,6 +106,76 @@ def before_save(self):
 		if self.team in teams or self.app in apps:
 			self.status = "Approved"
 
+	def _has_auto_deploy_marker(self) -> tuple[bool, str | None]:
+		"""<deploy-marker>-<bench-group> | <deploy-marker>"""
+		deploy_marker = frappe.db.get_single_value("Press Settings", "deploy_marker", cache=True)
+
+		if not self.message or deploy_marker not in self.message:
+			return False, None
+
+		bench_group = self.message.split(deploy_marker)[-1]
+		bench_group = bench_group.replace("-", "", 1) if bench_group else None
+
+		return True, bench_group
+
+	def _validate_bench_group(self, bench_group: str) -> bool:
+		"""Check if bench group exists and belongs to the team"""
+		try:
+			bench_group_doc: "ReleaseGroup" = frappe.get_doc("Release Group", bench_group)
+			return bench_group_doc.team == self.team
+		except frappe.DoesNotExistError:
+			return False
+
+	def _has_running_builds(self, bench_group: str) -> bool:
+		return bool(
+			frappe.get_value(
+				"Deploy Candidate Build",
+				{"status": ("in", ("Pending", "Running", "Scheduled")), "group": bench_group},
+			)
+		)
+
+	def _deploy_bench_group(self, bench_group: str) -> None:
+		"""Deploy bench group with this particular app update
+		1. Checks if the bench group already has running deploys
+		2. Checks if the bench group has the app that is being deployed
+		3. Deploy candidate can be created from the release group
+		"""
+
+		if self._has_running_builds(bench_group):
+			return
+
+		bench_group_has_app = frappe.db.get_value(
+			"Release Group App", {"parent": bench_group, "source": self.source}
+		)
+		if not bench_group_has_app:
+			return
+
+		group: "ReleaseGroup" = frappe.get_doc("Release Group", bench_group)
+		candidate: "DeployCandidate" = group.create_deploy_candidate(
+			apps_to_update=[app for app in group.apps if app.source == self.source]
+		)
+		if candidate:
+			candidate.schedule_build_and_deploy()
+
+	def trigger_deploy_via_commit_markers(self):
+		"""Check if the commit has deploy markers and trigger deploy"""
+		deploy, bench_group = self._has_auto_deploy_marker()
+
+		if not deploy:
+			return
+
+		if bench_group and not self._validate_bench_group(bench_group):
+			return
+
+		if bench_group:
+			self._deploy_bench_group(bench_group)
+
+		else:
+			for bench_group in frappe.db.get_all(
+				"Release Group", {"team": self.team, "enabled": 1}, pluck="name"
+			):
+				self._deploy_bench_group(bench_group)
```

Comment:
can we restrict it via any flag user/team can put on press?

so that we don't deploy everything they have

Reply:
so somehow put some checkbox or something (like a label mangesh mentioned as well) on dashboard - so that we can only deploy those benches - not all the team/user owns (?)

Reply:
Only deploying bench groups with the auto-deploy resource tag ([a8a22a6](https://github.com/frappe/press/pull/3966/commits/a8a22a6ebfb87f8d5de5ff304561b10a92337eb0))